gengokit/gengokit.go

```diff
@@ -160,134 +266,62 @@ func (g *generator) updateClientMethods(clientHandler io.Reader, svcFuncs []stri
 	currentFuncs := astMod.IndexFunctions()
 	code := astMod.Buffer()
 
-	err = g.applyTemplateForMissingMeths(clientMethodsTemplPath, currentFuncs, code)
+	// Get a templateExecutor that is identical but with only functions in the
+	// definition file and not in the previously generated file
+	trimmedTe, err := trimTemplateExecutorServiceFuncs(*te, currentFuncs)
 
+	out, err := applyTemplateFromPath(clientMethodsTemplPath, trimmedTe)
 	if err != nil {
 		return nil, errors.Wrap(err, "unable to apply client methods template")
 	}
 
+	code.ReadFrom(out)
+
 	return code, nil
 }
 
-// GenerateResponseFiles applies all template files for the generated
-// microservice and returns a slice containing each templated file as a
-// truss.NamedReadWriter.
-func (g *generator) GenerateResponseFiles(previousFiles []truss.NamedReadWriter, packageName string) ([]truss.NamedReadWriter, error) {
-	// Paths to handler files that may be modified programmatically
-	serverHandlerFilePath := "-service/handlers/server/server_handler.go"
-	clientHandlerFilePath := "-service/handlers/client/client_handler.go"
-
-	var codeGenFiles []truss.NamedReadWriter
-
-	// serviceFunctions is used later as the master list of all service methods
-	// which should exist within the handler files
-	var serviceFunctions []string
-	for _, meth := range g.templateExec.Service.Methods {
-		serviceFunctions = append(serviceFunctions, meth.GetName())
+// serviceFunctionNames returns a slice of function names which are in the
+// definition files plus the function "NewService". Used for inserting and
+// removing functions from previously generated handler files
+func serviceFunctionsNames(meths []*deftree.ServiceMethod) []string {
+	var svcFuncs []string
+	for _, meth := range meths {
```

Comment:
Something about having variables named `meths` and `meth` instead of `methods` and `m` seems kind of wrong to me. In an amusing way.